### ISSUE TITLE
Support `esphome.friendly_name` `Enhancement` 

### ISSUE THREAD

**Issue by twasilczyk** [Blackymas/NSPanel_HA_Blueprint#1719]
### Enhancement Summary

If I add `friendly_name` field to `esphome` section, NSPanel UI stops working

### Detailed Description

After adding `friendly_name` (e.g. to have `media-room-panel` as device name, but `Media room panel` as display name), device shows up in HA under this new name (as expected), but its sensors (and as I guess service names) are mixed up (see screenshot). It's probably because `nspanel_esphome.yaml` refer everywhere to `${name}` and not display name.

This in turn causes automation to try calling wrong service names:
> The automation "Media room panel configuration" (automation.media_room_panel_configuration) has an action that calls an unknown service: esphome.media_room_panel_media_room_panel_page_home.

### Additional Context

![friendly-name-problem](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/1107414/4b7687e6-f701-4fe0-a239-974d85dd6272)

My ESPHome yaml:
```
substitutions:
  device_name: media-room-panel
  wifi_ssid: !secret wifi_ssid
  wifi_password: !secret wifi_password
  ota_password: !secret ota_password

esphome:
  friendly_name: Media room panel

api:
  encryption:
    key: !secret nspanel_api_encryption_key

packages:
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: main
    files:
      - nspanel_esphome.yaml
    refresh: 300s

esp32:
  framework:
    type: esp-idf
```

**Comment by edwardtfn** [comment]
The issue you are seeing is related to us enforcing the device name on the entities, so we can use that to find service name.
The problem here is that when you use "Friendly name", that is automatically added to the entities names, but not to the services names. The blueprint doesn't have access to the list of services provided by a device, so we use the sensor names to find that and, when they are different, the blueprint will call invalid service name.

I've already started looking at that, but still needing some work. :(
I will see if I find some time for that.

**Comment by twasilczyk** [comment]
Do I understand correctly that you generate this name [here](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/nspanel_blueprint.yaml#L3948) from [this selector](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/nspanel_blueprint.yaml#L50)? Looks like you can [use device_attr](https://community.home-assistant.io/t/device-name-from-device-id-selector/400190) to pull device name (not friendly name).

I'm a little puzzled by a few things I see here though (I'm only reading code, didn't have a chance to run any tests):
- why device selector is called `nspanel_name` when it's [device ID, not name](https://www.home-assistant.io/docs/blueprint/selectors/#device-selector)?
- why `nspanel_name` generation is so complex and how it gets converted from Device ID to user friendly name?

For the suggestion - I tried renaming device after adding to HA, but it doesn't rename entities. This also breaks rendering entity names in device page (by default, friendly name is trimmed out in device page) - see screenshot in the first comment.

Don't feel rushed about this one - it's not as important as other bugs/FRs and it feels like low-ish hanging fruit for others to start contributing to this project.

**Comment by edwardtfn** [comment]
> Do I understand correctly that you generate this name [here](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/nspanel_blueprint.yaml#L3948) from [this selector](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/nspanel_blueprint.yaml#L50)? Looks like you can [use device_attr](https://community.home-assistant.io/t/device-name-from-device-id-selector/400190) to pull device name (not friendly name).

Nice! I will look at that.

**Comment by edwardtfn** [comment]
> why device selector is called nspanel_name when it's [device ID, not name](https://www.home-assistant.io/docs/blueprint/selectors/#device-selector)?

Until v3.1, that was a free text field for the device name, but it was one of the most common mistakes from users side. We changed for a device selector, but to avoid a breaking change, we kept the name and supported both selections for a while.

> why nspanel_name generation is so complex and how it gets converted from Device ID to user friendly name?

For supporting the legacy entries I've mentioned.

**Comment by edwardtfn** [comment]
> Don't feel rushed about this one - it's not as important as other bugs/FRs and it feels like low-ish hanging fruit for others to start contributing to this project.

I call this type of request as "relaxing changes"... When I'm too tired of working in something more complex, I spend some time in one of this relaxing thing, even if those are not that critical. 😉

**Comment by twasilczyk** [comment]
> For supporting the legacy entries I've mentioned.

Makes sense now. Worth adding a comment or two then.

**Comment by edwardtfn** [comment]
Done!
Will be included in the coming release.
Please feel free to test it in `dev`. You might have to reconnect your panel with Home Assistant like described in the last row of this table: https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/error_initializing.md#blueprint-is-not-detected

**Comment by edwardtfn** [comment]
We just released a patch with solves this: [v4.2.5 - Celebrating 1000 Stars with Enhanced Functionality and UI Improvements](https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v4.2.5)
Could you please update your system and let me know if the issue was solved?

**Comment by twasilczyk** [comment]
Thank you! Just verified and indeed the problem is solved.
